src/corona_hakab_model/infection.py

```diff
@@ -121,9 +155,22 @@ def _get_random_infection_info(self, agent_id: int, infection_probs: np.ndarray)
         connection_type = np.random.choice(len(infection_probs), p=infection_probs / np.sum(infection_probs))
 
         # determine infector
-        infectious_circle = \
-            [c for c in self.manager.social_circles_by_agent_index[agent_id] if c.connection_type == connection_type][0]
-        agents_id = [a.index for a in infectious_circle.agents]
+        agents_id = []
+
+        if connection_type in connection_types.With_Random_Connections:
+            infectious_circle = [circle for circle in self.manager.social_circles_by_agent_index[agent_id]
+                                 if circle.connection_type == connection_type][0]
+            agents_id = [agent.index for agent in infectious_circle.agents]
+
+        if connection_type in connection_types.With_Geo_Random_Connections:
+            agent_connection_type_circle = [circle for circle in self.manager.social_circles_by_agent_index[agent_id]
+                                            if circle.connection_type == connection_type][0]
+            agent_geo_circle = [geo_circle for geo_circle in self.manager.geographic_circles
+                                for circle in geo_circle.connection_type_to_social_circles[connection_type]
+                                if circle.guid == agent_connection_type_circle.guid][0]
+            infectious_circles = \
+                [circle for circle in agent_geo_circle.connection_type_to_social_circles[connection_type]]
+            agents_id = [agent.index for circle in infectious_circles for agent in circle.agents]
```

Comment:
i am not sure i understand why you did it this way?
why not use:
```
geo_circle = manager.geographic_circle_by_agent_index[agent_id]
agents_id = [agent.index for circle in geo_circle.connection_type_to_social_circles[connection_type] for agent in circle.agents]
```

Reply:
i did that in the beginning and understood that this is not correct, the agent can work in a social circle that is not in his own geo circle.
took my ages to find and solve this problem.

Reply:
good point.
won't the following be good enough?
`agent_geo_circle = [geo_circle for geo_circle in self.manager.geographic_circles 
                                 if agent_connection_type_circle in geo_circle.connection_type_to_social_circles[connection_type]][0]`

default comparison should be good enough here, and i don't think there is any reason to override it by comparing the .guid of the circles.

Reply:
done